### ISSUE TITLE
proxy not working in latest release - 

### ISSUE THREAD

**Issue by GameCrazi** [Ascoware/get-iplayer-automator#68]
---------------

*What steps will reproduce the problem? Be specific, and provide as much detail as possible.*

downloads report that outside of the uk - however proxy is in uk and works with 
1.8.7 get_iplayer_automator  

*What is the expected output? What do you see instead?*

*What version of GiA are you using? On what version of OS X?*
1.9.7 OSX 10.13 

*Are you using a web proxy? DNS proxy? VPN? VPS?*
yes 

*What program are you attempting to download (full title, series and episode)? **Provide only ONE program as an example.** *

ITV Liar 

*Please provide any additional information below.*

**Comment by GameCrazi** [comment]
Output from log

```
AppController: Downloads Finished
GetITVShows: ITV Cache Update Starting 
GetITVShows (Info): Merge complete B/F Programmes: 1354 C/F Programmes: 1328 Today Programmes: 170 
AppController: Index Updated.
GetITVShows: Update Finished


INFO: Loading proxy settings...
INFO: Custom Proxy: address=[http://droplet:9841] length=19
INFO: Proxy load complete.
INFO: Testing proxy (may take up to 30 seconds)...
INFO: Proxy test complete.
INFO: Using proxy: http://droplet:9841

AppController: Starting Downloads

Downloading Show 1/2:
ITV Download (ID=2/4547/0005#002): Downloading Liar - Mon Oct 09
INFO: Preparing Request for Auth Info
DEBUG: Metadata URL: http://mercury.itv.com/PlaylistService.svc
INFO: Requesting Metadata.
DEBUG: Metadata response status code: 200
DEBUG: Metadata response: <s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/">
   <s:Body>
      <GetPlaylistResponse xmlns="http://tempuri.org/">
         <GetPlaylistResult>
            <Fault xmlns="">
               <faultcode>InvalidGeoRegion</faultcode>
               <faultstring>The client is in a geographical region that does not have access to the content</faultstring>
            </Fault>
         </GetPlaylistResult>
      </GetPlaylistResponse>
   </s:Body>
</s:Envelope>

ITV Download (ID=2/4547/0005#002): ERROR: Access denied to users outside UK.
ITV Download (ID=2/4547/0005#002): Download Failed
```

**Comment by GameCrazi** [comment]
In ITVDownload.m 

I changed 

```
NSMutableDictionary *proxyDict = [NSMutableDictionary dictionaryWithObjectsAndKeys:
                                          (NSString *)kCFProxyTypeKey, (NSString *)kCFProxyTypeHTTP,
                                          (NSString *)kCFNetworkProxiesHTTPEnable, @(1),
                                          (NSString *)kCFNetworkProxiesHTTPProxy, self.proxy.host,
                                          (NSString *)kCFNetworkProxiesHTTPPort, @(self.proxy.port),
                                          nil];
```

to

```
        NSDictionary *proxyDict = @{
                                    @"HTTPEnable"  : [NSNumber numberWithInt:1],
                                    (NSString *)kCFStreamPropertyHTTPProxyHost  : self.proxy.host,
                                    (NSString *)kCFStreamPropertyHTTPProxyPort  : @(self.proxy.port),
                                    };
```

the above worked - 

digging more into why original didn't work .

**Comment by skovatch** [comment]
Nice! If you make a pull request out of that I can merge it in.

**Comment by skovatch** [comment]
Try this. That dictionary needs to be mutable because there may be a username and password to add.
```
        NSMutableDictionary *proxyDict = [NSMutableDictionary dictionaryWithObjectsAndKeys:
                                          (__bridge NSString *)kCFProxyTypeKey, (__bridge NSString *)kCFProxyTypeHTTP,
                                          @"HTTPEnable", @(1),
                                          (__bridge NSString *)kCFStreamPropertyHTTPProxyHost, self.proxy.host,
                                          (__bridge NSString *)kCFStreamPropertyHTTPProxyPort, @(self.proxy.port),
                                          nil];
```

**Comment by danaalk** [comment]
I couldnt download from BBC but ITV was fine with my UK proxy, same problem it said I was outside the UK

Downloading Show 7/9:
BBC Download (ID=b09b5154): Downloading Reformation - Part 2
INFO: tv episode PID detected (b09b5154)
INFO: Trying to download PID: b09b5154 using type: tv
INFO: PID found in cache: b09b5154
INFO: 1 matching programmes
WARNING: No media streams found for requested programme versions and recording modes.
WARNING: The BBC has blocked access to this programme because it has determined that you are using get_iplayer outside the UK.
BBC Download (ID=b09b5154): Reformation - Part 2 Failed

Downloading Show 8/9:
ITV Download (ID=2/4940/0001#001): Downloading Gordon Ramsay on Cocaine - Watch episodes - 2/4940/0001#001
INFO: Preparing Request for Auth Info
INFO: Requesting Metadata.
INFO: Metadata processed.
INFO: Requesting programme data.
INFO: Program data processed.
INFO: Launching RTMPDUMP...
RTMPDump v2.4-102-ga3a600d-get_iplayer
(c) 2010 Andrej Stepanchuk, Howard Chu, The Flvstreamer Team; license: GPL
Connecting ...
WARNING: HandShake: Type mismatch: client sent 6, server answered 8
INFO: Connected...
Starting download at: 0.000 kB
INFO: Metadata:
INFO:   duration              2714.88
INFO:   moovPosition          32.00
INFO:   width                 896.00
INFO:   height                504.00
INFO:   videocodecid          avc1
INFO:   audiocodecid          mp4a
INFO:   avcprofile            100.00
INFO:   avclevel              31.00
INFO:   aacaot                2.00
INFO:   videoframerate        25.00
INFO:   audiosamplerate       48000.00
INFO:   audiochannels         2.00
INFO: trackinfo:
INFO:   length                67872.00
INFO:   timescale             25.00
INFO:   language              und
INFO: sampledescription:
INFO:   sampletype            avc1
INFO:   length                130313216.00
INFO:   timescale             48000.00
INFO:   language              und
INFO: sampledescription:
INFO:   sampletype            mp4a

**Comment by skovatch** [comment]
Fixed in v1.9.8

**Comment by stickenhoffen** [comment]
I'm seeing issues with the latest version, GIA hangs at testing proxy. If I go back to 1.9.7 it's fine.
